utils/parallel.go

```diff
@@ -164,3 +164,49 @@ func RunInParallel(ctx context.Context, fs []SimpleFunc) (time.Duration, error)
 	wg.Wait()
 	return time.Since(start), bigError
 }
+
+// FloatFunc is for GetInParallel.
+type FloatFunc func(ctx context.Context) (float64, error)
+
+// GetInParallel runs all functions in parallel, return is elapsed time and n error.
+func GetInParallel(ctx context.Context, fs []FloatFunc) (time.Duration, []float64, error) {
+	start := time.Now()
+	ctx, cancel := context.WithCancel(ctx)
+	var floats []float64
+
+	var wg sync.WaitGroup
+
+	var bigError error
+	var bigErrorMutex sync.Mutex
+	storeError := func(float float64, err error) {
+		bigErrorMutex.Lock()
+		defer bigErrorMutex.Unlock()
+		if bigError == nil || !errors.Is(err, context.Canceled) {
+			bigError = multierr.Combine(bigError, err)
+		}
+	}
+
+	helper := func(f FloatFunc) {
+		defer func() {
+			if thePanic := recover(); thePanic != nil {
+				storeError(0, fmt.Errorf("got panic running something in parallel: %v", thePanic))
+				cancel()
+			}
+			wg.Done()
+		}()
+		float, err := f(ctx)
+		if err != nil {
+			storeError(float, err)
+			cancel()
+		}
+		floats = append(floats, float)
+	}
+
+	for _, f := range fs {
+		wg.Add(1)
+		go helper(f)
```

Comment:
this function is just a slightly edited version of `RunInParallel` so for consistency I kept everything the same accept adding the returns that are needed.

Reply:
I totally missed that this was mostly copied-and-pasted from other code! I wonder if it's possible to use generics to unify the two, since it sure looks to me like they should both be changed to use `PanicCapturingGoWIthCallback`, and it's annoying to fix the same bug multiple times over. Looking in the history, [Eric had the same opinion](https://github.com/viamrobotics/rdk/pull/766/files#r868188637), and Eliot just ignored him and marked the comment as resolved anyway. I wonder what happened in there.

If you want to keep it as-is for consistency with `RunInParallel`, I guess that's okay. but I do suspect that at some point in the future, someone is going to want to update `RunInParallel` and anything made from a copy of it.

Reply:
I did try to just modify `RunInParallel` at first but it's pretty specific so I added a new function, and now we basically have a get and a set function for the same thing which makes sense to me. I am going to keep the pattern consistent for now especially since it was previously decided on. if it comes up again in the future I am not opposed to changing them but it seems a bit unrelated to this PR